integration/tests/iam_v2_no_legacy_to_v2_force_upgrade_diagnostics.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+#shellcheck disable=SC2034
+test_name="iam_force_upgrade_to_v2_with_no_legacy"
+test_upgrades=true
+test_upgrade_strategy="none"
+test_diagnostics_pre_upgrade_filters="~skip-for-deep-upgrade"
+
+# Note: we can't run diagnostics AND inspec, so we don't include any inspec tests
+test_skip_diagnostics=false
```

Comment:
So here we are saying to run "diagnostics" instead of inspec. Not being familiar with diagnostics, could you elaborate on what this is doing, what test are being run if not inspec tests, etc. Thanks!